### DISCUSSION TITLE
only a few numbers are stored in IOBROKER

### DISCUSSION THREAD

**Discussion by otomabaer** [john30/ebusd#1088] [Q&A]
I'm a ebus beginner but I spent already lots of time reading and testing. So this is my last hope.

HW root@WP-Logger64lite:/home/ras# ebusctl scan result
08;Vaillant;HMU00;0607;5103
15;Vaillant;70000;0614;6903;21;22;33;0020266797;0082;050728;N2
76;Vaillant;VWZIO;0607;5103;21;21;49;0010023609;3110;007609;N3
ec;Vaillant;SOL00;0614;6903;21;22;33;0020266797;0082;050728;N2

My biggest problem is "only a few numbers are stored in IOBROKER"

ebusd config is: 
EBUSD_OPTS="--device=/dev/ttyUSB0 --scanconfig=full --port=8888 --accesslevel=* -l /var/log/ebusd.log --httpport=8889 --htmlpath=/var/ebusd/html"

I added latest logfile not to confuse you by misleading questions:
[ebusd.log](https://github.com/john30/ebusd/files/13653145/ebusd.log)

its my first ask for help the community. Interested to see what will happen.
thanks in advance

**Comment by otomabaer** [comment]
I changed HW from ESERA to John's adapter. I really impressed by size, function and price => good job!

But I still have same problem ONLY a few parameter are listed calling: EBUSCTL I
Output: version: ebusd 23.2.23.2
update check: version 23.3 available
device: /dev/ttyAMA0, enhanced high speed, firmware 1.1[3c17].1[3c17]
access: *
signal: acquired
...

So, I'm still looking for help ;)

**Comment by otomabaer** [comment]
Reading single values is possible!

root@WP-Logger64lite:/home/ras# ebusctl
localhost: r -c hmu hours
1947

**Comment by otomabaer** [comment]
After read a parameter manually by "ebusctl"  e. g. "r -c hmu hours" this parameter appears in "ebusctl find -d":

root@WP-Logger64lite:/home/ras# ebusctl find -d
broadcast outsidetemp = 7.691
broadcast vdatetime = 10:39:10;04.01.2024
**hmu Hours = 1947**
hmu SetMode = auto;41.0;-;-;0;1;1;0;0;0
hmu State = 0;128;193;8
hmu Status01 = 42.5;39.0;-;-;-;on
hmu StatusCirPump = off
hmu WaterThroughput = 733
sc YieldThisYear =  (ERR: invalid position for 31ecb509030d3f00 / 00)
scan.08  = Vaillant;HMU00;0607;5103
scan.08 id = 21;22;12;0010021118;0001;006414;N9
scan.15  = Vaillant;70000;0614;6903
scan.15 id = 21;22;33;0020266797;0082;050728;N2
scan.76  = Vaillant;VWZIO;0607;5103
scan.76 id = 21;21;49;0010023609;3110;007609;N3
scan.ec  = Vaillant;SOL00;0614;6903
scan.ec id = 21;22;33;0020266797;0082;050728;N2

**Comment by john30** [comment]
no idea how this works in iobroker. guess you need to crawl the appropriate forums there.
as a workaround (until you know how this is supposed to work there) you could just use e.g. a cronjob to regularly query the values you want. or use the polling feature with ebusd

**Reply by otomabaer** [reply]
thx for your attention! try to keep my answer short.

1) you are right with polling.
   If I start "readall.sh", I get ALL (guess more than 100) values available - all "ec-data" are not available.

2) If I start "ebusctl find -d"
   a)    **BEFORE**  "read.sh",  I get the few (shown above 6 values)
   b)    **AFTER**     "read.sh", ALL (guess more than 100) values available - all "ec-data" are not available

My question: Are the CSV files used to create read commands OR  is the daemon purley listening and CSVs are used to decode received messages?

PS: I had a short look on the code. I'm pressed. I guess 3 man years effort and well done!!!
